node/src/main/kotlin/net/corda/node/services/statemachine/FlowStateMachineImpl.kt

```
@@ -258,12 +258,15 @@ class FlowStateMachineImpl<R>(override val id: StateMachineRunId,
 
         openThreadLocalWormhole()
         setLoggingContext()
-        initialiseFlow()
 
         logger.debug { "Calling flow: $logic" }
         val startTime = System.nanoTime()
+        var initialised = false
         val resultOrError = try {
 
+            initialiseFlow()
+            initialised = true
```

Minor one but would it be bit prettier if we make  `initialiseFlow()` return boolean value instead? 
then `initialised = initialiseFlow()`

---

Returning`true` from `initialiseFlow` was not done because there is no way to make it return false as it will be throwing an exception rather than returning false.

Meaning that it will either return true or an exception. Returning it from the function implies there is a way for it to return false.